### PR TITLE
Setup with docker

### PR DESCRIPTION
**Pull request by ViktorBarzin**
**TLDR: Install docker-compose, run `docker-compose up,` save tons of setup time**

Stop worrying about dependencies and environments and focus on important stuff.
Setup the entire project with 1 command. Running ` docker-compose up` does the following:

1. Pulls the official images for python3.6 and postgres.
2. Setups a postgres instance with the provided in the `environment` section credentials
3. Runs the webapp, runs migrations and initial seed as specified in `bootstrap.sh` 
4. Mounts the source code as a volume in the container so any changes you make will affect the container immediately.(Yes it will restart the server as it does usually)
To forcefully restart the webserver run `docker-compose restart webapp`
5. Runs the webserver on localhost:8000 as usual without exposing any other ports.

**Notes**

- Db must be running before the webapp. However, docker does not provide a solution for this yet so I have added a 7-8 seconds wait for the webapp container to make sure the initial db setup is done before trying to connect to it. If you have any troubles, just rerun `docker-compose up` and the db will just start rather build from scratch and problem will be fixed.
- You must install bower_components from the host because the container does not have write access on the mounted volume.

If you like it I can setup a production script where the real fun begins. 
Cheers ^_^

### CODE REVIEW

**Comment by RadoRado** [comment]
@ViktorBarzin we are going to take a look at that since we are doing some e2e grader tests with docker - github.com/HackSoftware/HackGraderTests

Thanks for the input :beers:
